controllers/oneagent/oneagent_controller.go

```diff
@@ -98,7 +99,7 @@ func (r *ReconcileOneAgent) Reconcile(ctx context.Context, rec *utils.Reconcilia
 		}
 	}
 
-	if rec.IsOutdated(r.instance.Status.OneAgent.LastHostsRequestTimestamp, updInterval) {
+	if !r.instance.FeatureDisableHostsRequests() && rec.IsOutdated(r.instance.Status.OneAgent.LastHostsRequestTimestamp, updInterval) {
```

Comment:
Following the execution path, this check is in place solely for preventing updates of `OneAgentInstance` in the status field. So:
1. How about removing `OneAgentInstance.Version` (it is documentation only anyways) and remove the condition?
2. Rething the use cases for `OneAgentInstance`, possibly remove it entirely?